app/src/lib/menu-update.ts

```diff
@@ -164,12 +166,29 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.disable('pull')
   }
 
+  const welcomeScopedIds: ReadonlyArray<MenuIDs> = [
+    'new-repository',
+    'add-local-repository',
+    'clone-repository',
+    'create-branch',
+  ]
+
+  if (inWelcomeFlow) {
+    for (const id of welcomeScopedIds) {
+      menuStateBuilder.disable(id)
+    }
+  } else {
+    for (const id of welcomeScopedIds) {
+      menuStateBuilder.enable(id)
```

Comment:
We do, but I believe we still need the logic for flipping `repositoryActive` to false if we're showing the welcome flow since it's possible that we have an active repository selection _and_ are showing the welcome flow at the same time.